service/worker/scheduler/workflow.go

```diff
@@ -409,7 +411,15 @@ func (s *scheduler) processPatch(patch *schedulepb.SchedulePatch) {
 	s.logger.Debug("Schedule patch")
 
 	if trigger := patch.TriggerImmediately; trigger != nil {
-		now := s.now()
+		var now time.Time
+		if s.hasMinVersion(TriggerImmediatelyTimestamp) {
+			now = timestamp.TimeValue(trigger.ScheduledTime)
+			if now.IsZero() {
+				now = s.now()
+			}
+		} else {
+			now = s.now()
+		}
```

Comment:
could simplify to
```suggestion
		now := s.now()
		if t := timestamp.TimeValue(trigger.ScheduledTime); !t.IsZero() && s.hasMinVersion(TriggerImmediatelyTimestamp) {
			now = t
		}
```
but it looks correct either way